bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/util/PidTranslationUtils.java

```diff
@@ -0,0 +1,576 @@
+        mapDescObj.put("0xF110", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_UNKNOWN, CATEGORY_UNDEFINED, "", "",
+                "pid-info-heartbeat-delay", "pid-info-heartbeat-delay-text"));
+        mapDescObj.put("0xFF01", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_UNKNOWN, CATEGORY_UNDEFINED, "", "",
+                "pid-measured-temperature-1", "pid-measured-temperature-1-text"));
+        mapDescObj.put("0xFF02", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_UNKNOWN, CATEGORY_UNDEFINED, "", "",
+                "pid-measured-temperature-2", "pid-measured-temperature-2-text"));
+        mapDescObj.put("0xFF03", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_UNKNOWN, CATEGORY_UNDEFINED, "", "",
+                "pid-measured-temperature-3", "pid-measured-temperature-3-text"));
+        mapDescObj.put("0xFF04", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_UNKNOWN, CATEGORY_UNDEFINED, "", "",
+                "pid-measured-temperature-4", "pid-measured-temperature-4-text"));
+        mapDescObj.put("0x061A", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_DECIMAL, CATEGORY_PRESSURE, "", "",
+                "pid-air-quality-pressure-value", "pid-air-quality-pressure-value-text"));
+        mapDescObj.put("0x061B", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-co2-value", "pid-air-quality-co2-value-text"));
+        mapDescObj.put("0x061C", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-co-value", "pid-air-quality-co-value-text"));
+        mapDescObj.put("0x061D", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-no2-value", "pid-air-quality-no2-value-text"));
+        mapDescObj.put("0x061E", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-o3-value", "pid-air-quality-o3-value-text"));
+        mapDescObj.put("0x061F", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-pm10-value", "pid-air-quality-pm10-value-text"));
+        mapDescObj.put("0x0620", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-pm25-value", "pid-air-quality-pm25-value-text"));
+        mapDescObj.put("0x0621", createFreeAtHomePairingIdTranslation(PID_VALUETYPE_INTEGER, CATEGORY_UNDEFINED, "", "",
+                "pid-air-quality-voc-value", "pid-air-quality-voc-value-text"));
+
+        MAP_TRANSLATOR = Collections.unmodifiableMap(mapDescObj);
+    }
+
+    public static String getShortTextForPairingId(String Key) {
+        PIdContainerClass desc = MAP_TRANSLATOR.get(Key);
+
+        return (desc != null) ? desc.label : "Unknown label";
```

Comment:
Is this expected to happen (key not being found) - and in which scenario? Since all the other methods would also fail to find it, it seems the end result will be rather strange.

Reply:
The free@home system is having huge number of different function and pairing IDs. The table is implemented with care, however I cannot test all combinations. Therefore if an invalid key is given and the String is not found, at least some visible text will be displayed that the user can report it.

Reply:
Perhaps it would be better to throw a dedicated exception and log this as a warning elsewhere?

Reply:
OK... I will look for a solution

Reply:
**Fixed**

exception handling and logging is implemented